qt/applications/workbench/workbench/utils/test/test_recentlyclosedscriptsmenu.py

```diff
@@ -0,0 +1,269 @@
+        super().__init__()
+
+
+class MockQObject(QObject):
+    def __init__(self):
+        super().__init__()
+        self.editor = MagicMock()
+
+
+@start_qapplication
+class RecentlyClosedScriptsMenuTest(TestCase):
+    def tearDown(self):
+        if os.path.isdir(working_directory):
+            shutil.rmtree(working_directory)
+
+    @patch("workbench.utils.recentlyclosedscriptsmenu.CONF", new_callable=MockCONF)
+    @patch.object(RecentlyClosedScriptsMenu, 'repopulate_menu', return_value=None)
+    def test_repopulate_menu_is_called_on_aboutToShow_signal_being_triggered(self, repopulate_menu, _):
+        menu = RecentlyClosedScriptsMenu(None)
+
+        menu.aboutToShow.emit()
+
+        repopulate_menu.assert_called_once()
+
+    @patch("workbench.utils.recentlyclosedscriptsmenu.CONF", new_callable=MockCONF)
+    @patch.object(RecentlyClosedScriptsMenu, 'populate_menu', return_value=None)
+    def test_repopulate_menu_clears_and_then_populates(self, populate_menu, _):
+        fake_object = FakeQObject()
+        menu = RecentlyClosedScriptsMenu(None)
+        menu.addAction(QAction(fake_object))
+
+        self.assertEqual(len(menu.actions()), 1)
+
+        menu.repopulate_menu()
+
+        self.assertEqual(len(menu.actions()), 0)
+        populate_menu.assert_called_once()
+
+    @patch("workbench.utils.recentlyclosedscriptsmenu.CONF", new_callable=MockCONF)
+    @patch.object(RecentlyClosedScriptsMenu, 'load_scripts_from_settings', return_value=[])
+    def test_populate_menu_adds_a_default_if_no_scripts_found(self, _, __):
+        fake_object = FakeQObject()
+        menu = RecentlyClosedScriptsMenu(fake_object)
+
+        self.assertEqual(len(menu.actions()), 0)
+
+        menu.populate_menu()
+
+        self.assertEqual(len(menu.actions()), 1)
+        self.assertEqual(menu.actions()[0].text(), "No recently closed scripts found")
+
+    @patch("workbench.utils.recentlyclosedscriptsmenu.CONF", new_callable=MockCONF)
+    @patch.object(RecentlyClosedScriptsMenu, 'load_scripts_from_settings', return_value=["1", "2", "3"])
+    def test_populate_menu_adds_same_number_of_actions_as_script_paths(self, _, __):
+        fake_object = FakeQObject()
+        menu = RecentlyClosedScriptsMenu(fake_object)
+
+        self.assertEqual(len(menu.actions()), 0)
+        menu.populate_menu()
+
+        self.assertEqual(len(menu.actions()), 3)
+        self.assertEqual(menu.actions()[0].text(), "1")
+        self.assertEqual(menu.actions()[1].text(), "2")
+        self.assertEqual(menu.actions()[2].text(), "3")
+
+    @patch("workbench.utils.recentlyclosedscriptsmenu.CONF", new_callable=MockCONF)
+    def test_size_path_correctly_shortens_too_long_string(self, _):
+        menu = RecentlyClosedScriptsMenu(None)
+        path = "12345678912345678912345678912345678912845176451762534716412786459176549671549673"
+
+        self.assertEqual("...45678912845176451762534716412786459176549671549673", menu.size_path_correctly(path))
```

Comment:
This would need a change if you decide to alter the string shortening as above

Reply:
Good note thank you.